plugin-server/src/cdp/services/messaging/mailjet-manager.service.test.ts

```diff
@@ -0,0 +1,256 @@
+                path: '/',
+                hostname: 'localhost',
+                host: 'localhost',
+                fresh: false,
+                stale: true,
+                xhr: false,
+                cookies: {},
+                method: 'POST',
+                params: {},
+                query: {},
+                route: {},
+                secure: false,
+                signedCookies: {},
+                originalUrl: '/',
+                url: '/',
+                baseUrl: '/',
+            } as unknown as express.Request & { rawBody?: Buffer }
+
+            const result = await service.handleWebhook(req)
+            expect(result.status).toBe(403)
+            expect(result.message).toBe('Invalid signature')
+        })
+
+        it('should process valid webhook events', async () => {
+            const timestamp = Date.now().toString()
+            const payload = JSON.stringify({
+                event: 'sent',
+                time: Date.now(),
+                email: 'test@example.com',
+                mj_campaign_id: 1,
+                mj_contact_id: 1,
+                message_id: 'test-message-id',
+                custom_id: 'test-custom-id',
+                payload: {},
+            })
+            const signature = crypto.createHmac('sha256', secretKey).update(`${timestamp}.${payload}`).digest('hex')
+
+            const req = {
+                headers: {
+                    'x-mailjet-signature': signature,
+                    'x-mailjet-timestamp': timestamp,
+                },
+                rawBody: Buffer.from(payload),
+                body: parseJSON(payload),
+                get: () => undefined,
+                header: () => undefined,
+                accepts: () => false,
+                acceptsCharsets: () => false,
+                acceptsEncodings: () => false,
+                acceptsLanguages: () => false,
+                param: () => undefined,
+                is: () => false,
+                protocol: 'http',
+                ip: '127.0.0.1',
+                ips: [],
+                subdomains: [],
+                path: '/',
+                hostname: 'localhost',
+                host: 'localhost',
+                fresh: false,
+                stale: true,
+                xhr: false,
+                cookies: {},
+                method: 'POST',
+                params: {},
+                query: {},
+                route: {},
+                secure: false,
+                signedCookies: {},
+                originalUrl: '/',
+                url: '/',
+                baseUrl: '/',
+            } as unknown as express.Request & { rawBody?: Buffer }
+
+            const result = await service.handleWebhook(req)
+            expect(result.status).toBe(200)
+            expect(result.message).toBe('OK')
+        })
+
+        it('should handle different event types', async () => {
+            const timestamp = Date.now().toString()
+            const events = [
```

Comment:
These cases should be `it.each(cases)` so that if any individual one fails its easier to see which one it was and debug it. We have a bunch of examples of this across the code